src/contracts/misc/GhoStewardV2.sol

```diff
@@ -0,0 +1,286 @@
+  mapping(uint256 => mapping(uint256 => address)) internal _gsmFeeStrategiesByRates;
+  EnumerableSet.AddressSet internal _gsmFeeStrategies;
+
+  /**
+   * @dev Only Risk Council can call functions marked by this modifier.
+   */
+  modifier onlyRiskCouncil() {
+    require(RISK_COUNCIL == msg.sender, 'INVALID_CALLER');
+    _;
+  }
+
+  /**
+   * @dev Only methods that are not timelocked can be called if marked by this modifier.
+   */
+  modifier notTimelocked(uint40 timelock) {
+    require(block.timestamp - timelock > MINIMUM_DELAY, 'DEBOUNCE_NOT_RESPECTED');
+    _;
+  }
+
+  /**
+   * @dev Constructor
+   * @param owner The address of the owner of the contract
+   * @param addressesProvider The address of the PoolAddressesProvider of Aave V3 Ethereum Pool
+   * @param ghoToken The address of the GhoToken
+   * @param riskCouncil The address of the risk council
+   */
+  constructor(address owner, address addressesProvider, address ghoToken, address riskCouncil) {
+    require(owner != address(0), 'INVALID_OWNER');
+    require(addressesProvider != address(0), 'INVALID_ADDRESSES_PROVIDER');
+    require(ghoToken != address(0), 'INVALID_GHO_TOKEN');
+    require(riskCouncil != address(0), 'INVALID_RISK_COUNCIL');
+
+    POOL_ADDRESSES_PROVIDER = addressesProvider;
+    GHO_TOKEN = ghoToken;
+    RISK_COUNCIL = riskCouncil;
+
+    _transferOwnership(owner);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateFacilitatorBucketCapacity(
+    address facilitator,
+    uint128 newBucketCapacity
+  ) external onlyRiskCouncil notTimelocked(_facilitatorsBucketCapacityTimelocks[facilitator]) {
+    require(_controlledFacilitatorsByAddress[facilitator], 'FACILITATOR_NOT_CONTROLLED');
+    (uint256 currentBucketCapacity, ) = IGhoToken(GHO_TOKEN).getFacilitatorBucket(facilitator);
+    require(
+      _isIncreaseLowerThanMax(currentBucketCapacity, newBucketCapacity, currentBucketCapacity),
+      'INVALID_BUCKET_CAPACITY_UPDATE'
+    );
+
+    _facilitatorsBucketCapacityTimelocks[facilitator] = uint40(block.timestamp);
+
+    IGhoToken(GHO_TOKEN).setFacilitatorBucketCapacity(facilitator, newBucketCapacity);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGhoBorrowCap(
+    uint256 newBorrowCap
+  ) external onlyRiskCouncil notTimelocked(_ghoTimelocks.ghoBorrowCapLastUpdate) {
+    DataTypes.ReserveConfigurationMap memory configuration = IPool(
+      IPoolAddressesProvider(POOL_ADDRESSES_PROVIDER).getPool()
+    ).getConfiguration(GHO_TOKEN);
+    uint256 currentBorrowCap = configuration.getBorrowCap();
+    require(
+      _isIncreaseLowerThanMax(currentBorrowCap, newBorrowCap, currentBorrowCap),
+      'INVALID_BORROW_CAP_UPDATE'
+    );
+
+    _ghoTimelocks.ghoBorrowCapLastUpdate = uint40(block.timestamp);
+
+    IPoolConfigurator(IPoolAddressesProvider(POOL_ADDRESSES_PROVIDER).getPoolConfigurator())
+      .setBorrowCap(GHO_TOKEN, newBorrowCap);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGhoBorrowRate(
```

Comment:
as i understand this function the rate can only ever be **increased**, never decreased - is this on purpose?

Reply:
Hello, I confirm we failed to be clear on the wording.

The intent is to have the ability to move **both** ways.

Reply:
I will update the implementation